registry/storage/driver/s3-aws/s3.go

```diff
@@ -1466,17 +1404,11 @@ func (w *writer) Write(p []byte) (int, error) {
 			}
 			defer resp.Body.Close()
 
-			// reset uploaded parts
-			w.parts = nil
-			w.ready.Clear()
+			w.reset()
 
-			n, err := w.ready.ReadFrom(resp.Body)
-			if err != nil {
+			if _, err := io.Copy(w.buf, resp.Body); err != nil {
```

Comment:
I had concerns about this - but the above condition guarantees(?) this data will be less than `minChunkSize`, which is tiny.